src/git.ts

```diff
@@ -0,0 +1,122 @@
+import * as github from "@actions/github";
+import { Octokit } from "@octokit/rest";
+import * as core from "@actions/core";
+import parseDiff, { File as DiffFile } from "parse-diff";
+import { ReviewComment } from "./types";
+
+const GITHUB_TOKEN = core.getInput("GITHUB_TOKEN");
+const octokit = new Octokit({ auth: GITHUB_TOKEN });
+
+export async function getDiff(
+  owner: string,
+  repo: string,
+  pull_number: number,
+  action: string
+) {
+  // Do a full diff on new PRs, but do an incremental diff on updates
+  if (action === "opened") {
+    const response = await octokit.pulls.get({
+      owner,
+      repo,
+      pull_number,
+      mediaType: { format: "diff" },
+    });
+    return response.data;
+  } else if (action === "synchronize") {
+    const { data: eventData } = await octokit.pulls.get({
+      owner,
+      repo,
+      pull_number,
+    });
+    const newBaseSha = eventData.base.sha;
+    const newHeadSha = eventData.head.sha;
+
+    const response = await octokit.repos.compareCommits({
+      owner,
+      repo,
+      base: newBaseSha,
+      head: newHeadSha,
+      mediaType: { format: "diff" },
+    });
+    return response.data;
+  } else {
+    console.log("Unsupported event:", action);
+    return null;
```

Comment:
Returning null for unsupported events may lead to unexpected behavior in the calling function. Consider throwing an error instead to enforce better error handling.